src/app/public/modules/text-highlight/text-highlight.directive.ts

```diff
@@ -149,12 +149,6 @@ export class SkyTextHighlightDirective
   }
 
   private static cleanRegex(regex: string) {
-    return regex
-      .replace(/\\/g, '\\\\')
-      .replace(/\(/g, '\\(')
-      .replace(/\)/g, '\\)')
-      .replace(/\{/g, '\\{')
-      .replace(/\}/g, '\\}')
-      .replace(/\./g, '\\.');
+    return regex.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
```

Comment:
Mind if we add a unit test to check against all characters in `\^$*+?.()|{}[]`?

https://github.com/blackbaud/skyux-indicators/blob/master/src/app/public/modules/text-highlight/text-highlight.directive.spec.ts#L149-L171

Reply:
Sure thing. Done!